ambari-server/src/main/java/org/apache/ambari/server/view/ViewExtractor.java

```diff
@@ -201,9 +203,35 @@ private ClassLoader getArchiveClassLoader(ViewConfig viewConfig, File archiveDir
       urlList.add(classesDir.toURI().toURL());
     }
 
+        // include libs in additional classpath
+    for (File file : viewsAdditionalClasspath) {
+      if (file.isDirectory()) {
+        // add all files inside this dir.
+        addDirToClasspath(urlList, file);
+      } else if (file.isFile()) {
+        urlList.add(file.toURI().toURL());
+      }
+    }
+
     // include any libraries in the lib directory
     String libPath = archivePath + File.separator + ARCHIVE_LIB_DIR;
-    File   libDir  = archiveUtility.getFile(libPath);
+    File libDir = archiveUtility.getFile(libPath);
+    addDirToClasspath(urlList, libDir);
+
+    // include the archive directory
+    urlList.add(archiveDir.toURI().toURL());
+
+    LOG.trace("classpath for view {} is : {}", viewConfig.getName(), urlList);
+    return new ViewClassLoader(viewConfig, urlList.toArray(new URL[urlList.size()]));
+  }
+
+  /**
+   * Add all the files in libDir to urlList ignoring directories.
+   * @param urlList
+   * @param libDir
+   * @throws MalformedURLException
```

Comment:
`@param`/`@throws` without description is invalid, please either add descriptions or remove all 3 tags.

Reply:
added description for @param and removed @throws